### PR TITLE
Add smithing item helpers

### PR DESCRIPTION
**Pull request by kennytv**
because I can only count to 1

### CODE REVIEW

**Comment by CanadianBaconBoi** [comment]
All this does is add bulk code, if I were you, I'd learn to count to 1

**Comment by Proximyst** [comment]
There is already a patch for this exact thing in other inventories. This only adds more of the same as every other part of that same patch to the API.

**Review by Proximyst — DISMISSED** [review]

File: Spigot-API-Patches/0201-Add-item-slot-convenience-methods.patch [hunk 1]
```diff
@@ -177,6 +177,80 @@ index 4a0c43acc2714e095973eb78536041bb1a179ddc..acf2244f77133df53eb5f862c8e713c8
...
++public interface SmithingInventory extends Inventory {
++
++    // Paper start
++    /**
++     * Gets the input gear (first slot).
++     *
++     * @return input gear item
++     */
++    @Nullable
++    default ItemStack getInputGear() {
```
Comment (kashike) [inline Spigot-API-Patches/0201-Add-item-slot-convenience-methods.patch]: `getInputEquipment`?

File: Spigot-API-Patches/0201-Add-item-slot-convenience-methods.patch [hunk 2]
```diff
@@ -177,6 +177,80 @@ index 4a0c43acc2714e095973eb78536041bb1a179ddc..acf2244f77133df53eb5f862c8e713c8
...
++        return getItem(0);
++    }
++
++    /**
++     * Sets the input gear (first slot).
++     *
++     * @param itemStack item to set
++     */
++    default void setInputGear(@Nullable ItemStack itemStack) {
```
Comment (kashike) [inline Spigot-API-Patches/0201-Add-item-slot-convenience-methods.patch]: `setInputEquipment`?

**Review by kashike — CHANGES_REQUESTED** [review]

**Review by Proximyst — APPROVED** [review]

**Review by kashike — APPROVED** [review]

**Pull request merged**